### DISCUSSION TITLE
Checkpoints not getting saved

### DISCUSSION THREAD

**Discussion by nrjvarshney** [Lightning-AI/pytorch-lightning#7745] [code help: NLP / ASR / TTS] [answered]
I'm using the below callback 
`checkpoint_callback = pl.callbacks.ModelCheckpoint(
        dirpath=other_arguments.output_dir,
        monitor="val_loss",
        save_top_k=other_arguments.save_top_k,
        save_last=other_arguments.save_last,
        mode='min'
    )

    train_params = dict(
        accumulate_grad_batches=other_arguments.gradient_accumulation_steps,
        gpus=training_arguments.n_gpu,
        deterministic=True,
        max_epochs=other_arguments.num_train_epochs,
        precision=16 if training_arguments.fp_16 else 32,
        amp_level=training_arguments.opt_level,
        gradient_clip_val=training_arguments.max_grad_norm,
        checkpoint_callback=checkpoint_callback,
        fast_dev_run=other_arguments.do_fast_dev_run,
    )`

The output dir is empty which means the checkpoints are not getting saved. There is no error as well

**Comment by ananthsub** [comment, marked as answer]
Pass `checkpoint_callback` to the `callbacks` argument. `checkpoint_callback` is a boolean only flag to indicate whether checkpointing should be enabled or not.

**Reply by nrjvarshney** [reply]
Thanks, it is working now.
